applications/GeoMechanicsApplication/custom_constitutive/truss_backbone_constitutive_law.cpp

```diff
@@ -0,0 +1,166 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+// Project includes
+#include "truss_backbone_constitutive_law.h"
+#include "geo_mechanics_application_variables.h"
+#include "includes/properties.h"
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer TrussBackboneConstitutiveLaw::Clone() const
+{
+    return Kratos::make_shared<TrussBackboneConstitutiveLaw>(*this);
+}
+
+void TrussBackboneConstitutiveLaw::GetLawFeatures(Features& rFeatures)
+{
+    // Set the strain size
+    rFeatures.mStrainSize = 1;
+
+    // Set the space dimension
+    rFeatures.mSpaceDimension = 3;
+}
+
+double& TrussBackboneConstitutiveLaw::GetValue(const Variable<double>& rThisVariable, double& rValue)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        rValue = mAccumulatedStrain;
+    } else {
+        KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+    }
+    return rValue;
+}
+
+array_1d<double, 3>& TrussBackboneConstitutiveLaw::GetValue(const Variable<array_1d<double, 3>>& rThisVariable,
+                                                            array_1d<double, 3>& rValue)
+{
+    KRATOS_ERROR << "Can't get the specified value for " << rThisVariable << std::endl;
+}
+
+void TrussBackboneConstitutiveLaw::SetValue(const Variable<double>& rThisVariable,
+                                            const double&           rValue,
+                                            const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        mAccumulatedStrain = rValue;
+    } else {
+        KRATOS_ERROR << "Can't set the specified value for " << rThisVariable << std::endl;
+    }
+}
+
+double& TrussBackboneConstitutiveLaw::CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+                                                     const Variable<double>&      rThisVariable,
+                                                     double&                      rValue)
+{
+    if (rThisVariable == TANGENT_MODULUS) {
+        const auto youngs_modulus = rParameterValues.GetMaterialProperties()[YOUNG_MODULUS];
+        const auto axial_strain(rParameterValues.GetStrainVector()[0]);
```

Comment:
To be consistent with the previous line, I would rewrite this one as follows:
```suggestion
        const auto axial_strain = rParameterValues.GetStrainVector()[0];
```

Reply:
Done.